src/_nebari/stages/base.py

```diff
@@ -253,8 +265,24 @@ def render(self) -> Dict[pathlib.Path, str]:
             )
         }
         for root, dirs, filenames in os.walk(self.template_directory):
+            root_path = pathlib.Path(root)
+            if any(filename.endswith(".tf") for filename in filenames) and (
+                pathlib.PosixPath("modules")
+                in root_path.relative_to(self.template_directory).parents
+            ):
+                contents[
+                    pathlib.Path(
+                        self.stage_prefix,
+                        pathlib.Path.relative_to(
+                            root_path,
+                            self.template_directory,
+                        ),
+                    )
+                    / "_nebari.tf.json"
+                ] = opentofu.tf_render_objects(self._tf_objects_required_providers())
+
```

Comment:
If I understood correctly, this was included since we also wanted to render the _tf.json files for the module versions right? I've been testing this locally so far today, and even without this the actual versions have been correctly used by each stage and its underlying modules. I will follow on what we discussed back then to refresh my head around why we went this route, but we might not need ti anymore :)